internal/driven/publisher/wa.go

```diff
@@ -44,6 +50,68 @@ func (n *WaPublisher) Publish(ctx context.Context, msg core.Message) error {
 	return nil
 }
 
+func (n *WaPublisher) GetLoginQrCode(ctx context.Context) (*core.QrCodeLogin, error) {
+	var rsp RespGetLoginQrCode
+
+	resp, err := n.HttpClient.R().
+		SetContext(ctx).
+		SetBasicAuth(n.Username, n.Password).
+		SetResult(&rsp).
+		SetError(&rsp).
+		Get(fmt.Sprintf("%v/app/login", n.WaApiBaseUrl))
+	if err != nil {
+		return nil, fmt.Errorf("unable to make http request: %w", err)
+	}
+	if resp.IsError() {
+		slog.Error("failed to get login qr code", slog.String("response", rsp.String()))
+		if rsp.IsSessionAlreadyActive() {
+			return &core.QrCodeLogin{
+				Session:    true,
+				QrDuration: 0,
+				QrLink:     "",
+			}, nil
+		}
+
+		return nil, fmt.Errorf("failed to get login qr code: %s", resp.String())
+	}
+
+	qrLink := rsp.Results.QrLink
+	if strings.HasPrefix(qrLink, fmt.Sprintf("http://%s:", WA_PUBLISHER_DOCKER_DOMAIN)) || strings.HasPrefix(qrLink, fmt.Sprintf("https://%s:", WA_PUBLISHER_DOCKER_DOMAIN)) {
+		qrLink = strings.Replace(qrLink, WA_PUBLISHER_DOCKER_DOMAIN, WA_PUBLISHER_LOCALHOST_DOMAIN, 1)
+	}
```

Comment:
I think we can do better here without any checking. Lets do it later in other PR